### PR TITLE
Added support for using password on connections

### PR DESCRIPTION
**Pull request by hegga**
I need support for using passwords for establishing a connection to the IRC server. This is a simple implementation for supporting connection passwords.

### CODE REVIEW

File: irc_bot/irc_bot.py [hunk 1]
```diff
@@ -163,6 +163,7 @@ def __init__(self, config):
         self.invite_message = config.get('invite_message')
         self.nickserv_password = config.get('nickserv_password')
         self.use_ssl = config.get('use_ssl', False)
+        self.use_password = config.get('use_password', None)
```
Comment (cvium) [inline irc_bot/irc_bot.py]: I'd prefer if it was just `config.get('password')`

Comment (hegga) [reply irc_bot/irc_bot.py]: OK, I will delete this pull request and create a new one.

Comment (cvium) [reply irc_bot/irc_bot.py]: You can just change this one.

Comment (hegga) [reply irc_bot/irc_bot.py]: Ah! I will try that. This is my first pull request ever. Would you like me to change `self.use_password` to `self.password` as well?

Comment (cvium) [reply irc_bot/irc_bot.py]: Yeah, that'd be great.

File: irc_bot/irc_bot.py [hunk 2]
```diff
@@ -163,6 +163,7 @@ def __init__(self, config):
         self.invite_message = config.get('invite_message')
         self.nickserv_password = config.get('nickserv_password')
         self.use_ssl = config.get('use_ssl', False)
+        self.password = config.get('password', None)
```
Comment (cvium) [inline irc_bot/irc_bot.py]: The python dict `get()` method returns `None` if the key does not exist. It's therefore not necessary to do `config.get('password', None)`. `config.get('password')` is sufficient.

**Pull request merged**